pkg/operator2/secret.go

```diff
@@ -17,13 +18,37 @@ func (c *osinOperator) expectedSessionSecret() *v1.Secret {
 	secret, err := c.secrets.Secrets(targetName).Get(targetName, metav1.GetOptions{})
 	if err != nil || isInvalidSessionSecret(secret) {
 		glog.V(4).Infof("failed to get secret %s: %v", targetName, err)
-		return randomSessionSecret()
+		generatedSessionSecret := randomSessionSecret()
+		if isInvalidSessionSecret(generatedSessionSecret) {
+			panic("invalid session secret generated") // should never happen
```

Comment:
We shouldn't panic here. Return any errors and have the top-level sync function handle it.

Reply:
I am not really sure why we even have this check.  Seems like it should be in a unit test or a panic.

Reply:
yea, i see your point, ok